alsrb0504/20230425/미로 탈출 명령어/미로 탈출 명령어.js

```diff
@@ -0,0 +1,69 @@
+function solution(n, m, x, y, r, c, k) {
+  const FAIL = "impossible";
+  // d, l, r, u 순으로 우선 순위가 높음.
+  const [DOWN, LEFT, RIGHT, UP] = ["d", "l", "r", "u"];
+  const dir = {
+    d: [1, 0],
+    l: [0, -1],
+    r: [0, 1],
+    u: [-1, 0],
+  };
+  const answer = [];
+  let [curY, curX] = [x, y];
+  let [endY, endX] = [r, c];
+  let moveCnt = 0;
+
+  // 이동횟수가 k보다 작을때까지 반복
+  while (moveCnt < k) {
+    // dist: 현재 위치에서의 종료지점까지의 거리
+    // rest: 현재 더 움질일 수 있는 횟수
+    const dist = calcDist();
+    const rest = k - moveCnt;
+
+    // 종료 : 이동거리 부족 || 도달 불가능 (남은 여유 이동 가능거리가 2의 배수여야 함)
+    if (dist > rest || (dist < rest && (rest - dist) % 2 !== 0)) return FAIL;
+
+    // 1. 남은거리와 움직일 수 있는 횟수가 같은 경우
+    // => 최단 거리로 찾아가야 함
+    // => d, l, r, u 순으로 최단 거리 이동
+    if (dist === rest) {
+      let nextDir;
+
+      if (curY < endY) nextDir = DOWN;
+      else if (curX > endX) nextDir = LEFT;
+      else if (curX < endX) nextDir = RIGHT;
+      else nextDir = UP;
```

Comment:
```suggestion
      if (curY < endY)  nextDir = DOWN.repeat(endY-curY)
      else if (curX > endX) nextDir = LEFT;
      else if (curX < endX) nextDir = RIGHT;
      else nextDir = UP;
```

뭔가 이 부분은 repeat() 연산을 이용해서 시간을 줄일 수 있을 것 같기도 하네요...!

Reply:
헛 그렇네요 민규님 피드백 받고 좀 더 수정해봤습니다..!! ㅎㅎ 저렇게 코드를 짜면 while문의 반복을 줄일 수 있지 않을까 싶어요..!